pkg/git/github_test.go

```diff
@@ -0,0 +1,681 @@
+package git
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/go-github/v35/github"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestNewGitHubGITProvider(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "GitHubProvider")
+}
+
+var _ = Describe("NewGitHubGITProvider", func() {
+	var err error
+	var ctx context.Context
+
+	BeforeEach(func() {
+		err = nil
+		ctx = context.Background()
+	})
+
+	It("returns error when creating without url", func() {
```

Comment:
As commented in github.go if split in to a seperate function we can move these tests towards that function